Source/WebCore/dom/ElementAncestorIterator.h

```diff
@@ -56,10 +56,10 @@ class ElementAncestorRange {
     explicit ElementAncestorRange(ElementType* first);
     ElementAncestorIterator<ElementType> begin() const;
     static constexpr std::nullptr_t end() { return nullptr; }
-    ElementType* first() const { return m_first; }
+    ElementType* first() const { return m_first.get(); }
 
 private:
-    ElementType* const m_first;
+    RefPtr<ElementType> m_first;
```

Comment:
This code is performance-sensitive. I have started A/B tested for this for make sure this is OK.
Please hold off on landing this until perf results come back. In the future, staying out of WebCore and particularly dom/ and css/ folder would limit risk.